lib/tokenize.cpp

```diff
@@ -137,6 +137,10 @@ const Token * Tokenizer::isFunctionHead(const Token *tok, const std::string &end
         if (Token::Match(tok, "= 0|default|delete ;"))
             tok = tok->tokAt(2);
 
+        if (skipInitList)
+            while (Token::Match(tok, "[:,] %name% (|{")) // skip init list
```

Comment:
I have the feeling that base class initialization can contain :: and <>

    class Fred: public Foo::Bar {
        Fred() : Foo::Bar(123) {}
    };